client/src/SharedComponents/Slideshow/Slideshow.js

```diff
@@ -1,87 +1,94 @@
 import React, { Component } from "react";
-import { Slide } from "react-slideshow-image";
-import ProjectView from '../ProjectView'
 import StudentCard from '../StudentCard'
 import "./Slideshow.css";
+import Slider from "react-slick";
+import ProjectView from '../ProjectView'
 
 
-const dbProjects = ["","","","",""]
-const allProjects = dbProjects.map(project =>{
-return <ProjectView
-  projectImg={project}
-/>
-})
-
-const dbStudents = ["","","","",""]
-const allStudents = dbStudents.map(student =>{
-  return <StudentCard
-  studentImg= {student.img}
-   studentname={student.name}  />
-  })
-
-const properties = {
-  duration: 5000000000000000,
-  transitionDuration: 500,
-  infinite: false,
-  indicators: true,
-  arrows: true,
-};
-
 class Slideshow extends Component {
   state = {
-    projects: [],
-    students: []
+    projectSlide: [],
+    studentSlide: [],
   }
-  componentDidMount = () => {
-    let projectsSlide = [];
-    let studentsSlide = [];
-    for (let i = 0; i < allProjects.length; i += 3) {
-      projectsSlide.push(<div className="one-slide" key={Date.now()}>
-        {!allProjects[i]
-          ? <div></div>
-          : allProjects[i]
-        }
-        {!allProjects[i + 1]
-          ? <div></div>
-          : allProjects[i + 1]
-        }
-        {!allProjects[i + 2]
-          ? <div></div>
-          : allProjects[i + 2]
 
-        }
-      </div>)
-    }
-    for (let i = 0; i < allStudents.length; i += 3) {
-      studentsSlide.push(<div className="one-slide" key={Date.now()}>
-        <div className='student'>{allStudents[i]}</div>
-        <div className='student'>{allStudents[i + 1]}</div>
-        <div className='student'>{allStudents[i + 2]}</div>
-      </div>)
+  componentDidMount() {
+    const { projects, students } = this.props;
+
+    if (projects) {
```

Comment:
@Rahaf-96 I think it would be better if you moved this logic into render(), for a few reasons:
First, it's best to not store something in a component's state that is derived from its props.  It is better to just calculate derived data whenever you need it, instead of trying to calculate it and store it for later.  The reason is, it gets hard to keep the two things in sync (for example here, if the props change after the component is mounted, the state will not change, and then you aren't displaying the correct data any more).
Second, it's also best to not store JSX in component state.  You should only use state for simple data types like strings, numbers, arrays, objects, and so on.  If you put something like JSX in state, React sometimes can't detect if the JSX has changed.

Reply:
@tomduggan85 I moved all the logic into render(), could you check please if I did in the right way?

Reply:
Yes, it looks good to me!